pkg/stores/proxyalpha/proxy_store.go

```diff
@@ -0,0 +1,679 @@
+			}
+		}
+
+		resp, err := k8sClient.Patch(apiOp, id, pType, bytes, opts)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return resp, buffer, nil
+	}
+
+	resourceVersion := input.String("metadata", "resourceVersion")
+	if resourceVersion == "" {
+		return nil, nil, fmt.Errorf("metadata.resourceVersion is required for update")
+	}
+
+	opts := metav1.UpdateOptions{}
+	if err := decodeParams(apiOp, &opts); err != nil {
+		return nil, nil, err
+	}
+
+	resp, err := k8sClient.Update(apiOp, &unstructured.Unstructured{Object: moveFromUnderscore(input)}, metav1.UpdateOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	rowToObject(resp)
+	return resp, buffer, nil
+}
+
+// Delete deletes an object from a store.
+func (s *Store) Delete(apiOp *types.APIRequest, schema *types.APISchema, id string) (*unstructured.Unstructured, []types.Warning, error) {
+	opts := metav1.DeleteOptions{}
+	if err := decodeParams(apiOp, &opts); err != nil {
+		return nil, nil, nil
+	}
+
+	buffer := WarningBuffer{}
+	k8sClient, err := metricsStore.Wrap(s.clientGetter.TableClient(apiOp, schema, apiOp.Namespace, &buffer))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if err := k8sClient.Delete(apiOp, id, opts); err != nil {
+		return nil, nil, err
+	}
+
+	obj, _, err := s.byID(apiOp, schema, apiOp.Namespace, id)
+	if err != nil {
+		// ignore lookup error
+		return nil, nil, validation.ErrorCode{
+			Status: http.StatusNoContent,
+		}
+	}
+	return obj, buffer, nil
+}
+
+// ListByPartitions returns an unstructured list of resources belonging to any of the specified partitions
+func (s *Store) ListByPartitions(apiOp *types.APIRequest, schema *types.APISchema, partitions []partition.Partition) ([]unstructured.Unstructured, string, error) {
+	opts, err := listprocessor.ParseQuery(apiOp, s.namespaceCache)
+	if err != nil {
+		return nil, "", err
+	}
+	// warnings from inside the informer are discarded
+	buffer := WarningBuffer{}
+	client, err := s.clientGetter.TableAdminClient(apiOp, schema, "", &buffer)
+	if err != nil {
+		return nil, "", err
+	}
+	fields := getFieldsFromSchema(schema)
+	fields = append(fields, getFieldForGVK(attributes.GVK(schema))...)
+
+	inf, err := s.cacheFactory.CacheFor(fields, &tablelistconvert.Client{ResourceInterface: client}, attributes.GVK(schema), attributes.Namespaced(schema))
+	if err != nil {
+		return nil, "", err
+	}
+
+	list, continueToken, err := inf.ListByOptions(apiOp.Context(), opts, partitions, apiOp.Namespace)
+	if err != nil {
+		if errors.Is(err, informer.InvalidColumnErr) {
+			return nil, "", apierror.NewAPIError(validation.InvalidBodyContent, err.Error())
+		}
+		return nil, "", err
+	}
+
+	return list.Items, continueToken, nil
+}
+
+// WatchByPartitions returns a channel of events for a list or resource belonging to any of the specified partitions
+func (s *Store) WatchByPartitions(apiOp *types.APIRequest, schema *types.APISchema, wr types.WatchRequest, partitions []partition.Partition) (chan watch.Event, error) {
```

Comment:
@rmweir IIUC this is not excercised by UI code (yet), correct?

Reply:
[Its used in Watch of the partition store](https://github.com/rancher/steve/blob/vai-v2.9/pkg/stores/partitionalpha/store.go#L134).